dask/array/reductions.py

```diff
@@ -363,7 +364,11 @@ def partial_reduce(
         if len(out_chunks) == 0:
             meta = meta.sum()
         else:
-            meta = meta.reshape((0,) * len(out_chunks))
+            try:
+                if meta.size == 0:
+                    meta = meta.reshape((0,) * len(out_chunks))
+            except AttributeError:
+                meta = meta.reshape((0,) * len(out_chunks))
```

Comment:
Hrm, could you elaborate a bit on why this change is needed? (Sorry, it's not totally clear to me -- I might just be missing something)

Reply:
There were a couple failed tests related to trying to reshape an array size 1 to (0, 0) or (0, 0, 0) form. I believe this is because we now return [sys.maxsize] or [-sys.maxsize] instead of an empty array. It wouldn't make sense to try to reshape this arrays.

Reply:
I see, thanks for the extra information. In light of the suggestion in https://github.com/dask/dask/pull/9268#discussion_r934873831, perhaps this change isn't needed anymore since we would be returning an empty array? (I've not verified this myself, but it sounds like it could be the case)